packages/shared/src/components/sidebar/common.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { ReactNode, ReactElement } from 'react';
+import Link from 'next/link';
+import classed from '../../lib/classed';
+import { Button } from '../buttons/Button';
+import { SimpleTooltip } from '../tooltips/SimpleTooltip';
+import { TargetTypes } from '../utilities';
+import ArrowIcon from '../../../icons/arrow.svg';
+
+export interface SidebarMenuItems {
+  key: string;
+  items: SidebarMenuItem[];
+}
+
+export interface SidebarMenuItem {
+  icon: ReactElement;
+  title: string;
+  path?: string;
+  target?: TargetTypes;
+  action?: () => unknown;
+}
+
+export const ListIcon = ({
+  Icon,
+}: {
+  Icon: React.ComponentType<{ className }>;
+}): ReactElement => <Icon className="mr-3 w-5 h-5" />;
+export const ItemInner = ({
+  item,
+  openSidebar,
+}: {
+  item: SidebarMenuItem;
+  openSidebar: boolean;
+}): ReactElement => (
+  <>
+    {item.icon}
+    <span
+      className={`flex-1 text-left transition-opacity ${
+        openSidebar ? 'opacity-100' : 'opacity-0'
+      }`}
+    >
+      {item.title}
+    </span>
+  </>
+);
+export const btnClass = 'flex flex-1 items-center px-3 h-7';
+export const ButtonOrLink = ({
+  item,
+  children,
+}: {
+  item: SidebarMenuItem;
+  children?: ReactNode;
+}): ReactElement =>
```

Comment:
I suggest we always create an interface/type when it comes to props for `ReactElement` functions with 2 or more properties. It is quite a bit confusing to read sometimes also when updating the properties themselves or adding new ones.

Could be just my personal POV, but what do you think about that?

Reply:
That makes sense to me! let me work on that